test/helper.js

```diff
@@ -180,6 +210,65 @@ module.exports.payloadMethod = function (method, t) {
         t.error(err)
         t.strictEqual(response.statusCode, 422)
       })
+
+      sget({
+        method: upMethod,
+        url: 'http://localhost:' + fastify.server.address().port,
+        body: '',
+        headers: { 'Content-Type': 'application/json' },
+        timeout: 500
+      }, (err, response, body) => {
+        t.error(err)
+        t.strictEqual(response.statusCode, 422)
+      })
+    })
+
+    test(`${upMethod} returns 413 - Payload Too Large`, t => {
+      t.plan(upMethod === 'OPTIONS' ? 4 : 6)
+
+      sget({
+        method: upMethod,
+        url: 'http://localhost:' + fastify.server.address().port,
+        headers: {
+          'Content-Type': 'application/json',
+          'Content-Length': 1024 * 1024 + 1
+        }
+      }, (err, response, body) => {
+        t.error(err)
+        t.strictEqual(response.statusCode, 413)
+      })
+
+      // Node errors for OPTIONS requests with a stream body and no Content-Length header
+      if (upMethod !== 'OPTIONS') {
```

Comment:
It seems like there's a bug in Node. On Node 4 this test fails with an error and on the other versions it tries to attach too many `error` event handlers to the `req`, which produces a Node warning and errors with a `400` status.

Reply:
How many `error` handlers are attached to `req`?

Reply:
Here's what I'm getting from Node:

```
(node:10380) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 error listeners added. Use emitter.setMaxListeners() to increase limit
    at _addListener (events.js:280:19)
    at Socket.addListener (events.js:297:10)
    at Socket.Readable.on (_stream_readable.js:772:35)
    at Socket.socketOnWrap [as on] (_http_server.js:651:37)
    at Socket.socketOnError (_http_server.js:456:8)
    at onParserExecuteCommon (_http_server.js:467:19)
    at onParserExecute (_http_server.js:450:3)
```

I guess I meant the handlers are being attached to the socket, not `req` specifically.